### ISSUE TITLE
Hostname/IP does not match certificate's altnames

### ISSUE THREAD

**Issue by ScOut3R** [CrunchyData/postgres-operator#3161]
## Overview

Using the hostname provided by the cluster's secret my Node.js based TypeORM cluent is logging the following error

```
Error [ERR_TLS_CERT_ALTNAME_INVALID]: Hostname/IP does not match certificate's altnames: Host: [release name]-postgresql-cluster-primary.[namespace].svc. is not in the cert's altnames: DNS:[release-name]-postgresql-cluster-primary
```

By the looks of it the host provided by the secret does not match the domain names specified in the cluster's self-signed certificate.

In my understanding these two should match for the TLS connection to succeed.

## Environment

Please provide the following details:

- Platform: Kubernetes
- Platform Version: 1.22.8
- PGO Image Tag: ubi8-5.0.2-0
- Postgres Version: 13.4-0
- Storage: rook - ceph

**Comment by cbandy** [comment]
This is fixed in v5.0.3 by
- https://github.com/CrunchyData/postgres-operator/pull/2732

**Comment by ScOut3R** [comment]
Thank you @cbandy! I somehow missed this in the changelog. I updated the operator to 5.0.3 and I can confirm that it is working as expected.
